scripts/build-doc-list.js

```diff
@@ -7,20 +8,34 @@ const { slugify } = require('markdown-toc/lib/utils')
 const readingTime = require('reading-time')
 const { markdownToTxt } = require('markdown-to-txt')
 
-let specWeight = 100
-const result = []
+const yargs = require('yargs/yargs')
+const { hideBin } = require('yargs/helpers')
+const argv = yargs(hideBin(process.argv)).argv
+
 const basePath = 'pages'
-const postDirectories = [
+const defaultDirectoriesToProcess = [
   [`${basePath}/docs`, '/docs'],
   [`${basePath}/blog`, '/blog'],
   [`${basePath}/about`, '/about'],
-  [`${basePath}/jobs`, '/jobs'],
-]
-walkDirectories(postDirectories, result)
+  [`${basePath}/jobs`, '/jobs']
+];
+
+let specWeight = 100
+const result = []
+
+const output = argv.output;
+const groupCategoriesByFrontMatter = argv.groupByCategories || false;
```

Comment:
You don't need `|| false`. If `argv.groupByCategories` is false, that's it, you have it :) You may want to convert it to a boolean though:

```suggestion
const groupCategoriesByFrontMatter = !!argv.groupByCategories;
```